### PR TITLE
task(auth): Add DELETE /recovery-phone endpoint

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We want to provide the option for users to remove their recovery phone

## This pull request

- Adds a new endpoint DELETE `/recovery-phone`

## Issue that this pull request solves

Closes: FXA-10358

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/recovery-phone.ts [hunk 1]
```diff
@@ -117,6 +117,28 @@ class RecoveryPhoneHandler {
...
+      throw AppError.backendServiceFailure(
+        'RecoveryPhoneService',
+        'destroy',
+        { uid },
+        error
+      );
+    }
+
+    if (success) {
+      await this.glean.twoStepAuthPhoneRemove.confirmSubmit(request);
```
Comment (vbudhram) [inline packages/fxa-auth-server/lib/routes/recovery-phone.ts:137]: Looks like the backend event for this should be `two_step_auth_phone_remove_success` according to https://docs.google.com/spreadsheets/d/1DgohA2JG8gvFOjk_8zUHOi6DSDSPzqCCwoNVvirDpkA/edit?gid=1073857271#gid=1073857271

Comment (dschom) [reply packages/fxa-auth-server/lib/routes/recovery-phone.ts:137]: Thanks

File: packages/fxa-auth-server/lib/metrics/glean/index.ts [hunk 2]
```diff
@@ -325,6 +325,9 @@ export function gleanMetrics(config: ConfigType) {
         }
       ),
     },
+    twoStepAuthPhoneRemove: {
+      success: createEventFn('two_step_auth_phone_remove_remove_success'),
```
Comment (vbudhram) [inline packages/fxa-auth-server/lib/metrics/glean/index.ts:329]: Extra `remove`

**Review by vbudhram — APPROVED** [review]
@dschom Thanks, should be good with the one comment.

**Pull request merged**